staging/src/k8s.io/client-go/tools/cache/reflector.go

```diff
@@ -732,6 +732,9 @@ func watchHandler(start time.Time,
 	stopCh <-chan struct{},
 ) error {
 	eventCount := 0
+	initialEventsEndBookmarkWarningTickerDuration := 10 * time.Second
+	initialEventsEndBookmarkWarningTicker := newInitialEventsEndBookmarkTicker(clock, start, initialEventsEndBookmarkWarningTickerDuration, exitOnInitialEventsEndBookmark)
+	defer initialEventsEndBookmarkWarningTicker.stop()
```

Comment:
I would love to simplify this a bit to avoid polluting production code with debugging one that much.
I'm wondering if the following wouldn't be simpler:
1) Add NoopTicker type [ideally directly here: https://github.com/kubernetes/utils/blob/master/clock/clock.go#L168-L178 ]
2) Change ticker part to instead of handling nil there
3) change the interface of your wrapper to actually have: `warnIfNeeded()` that will do the logging
4) we need to stop the ticker after initial event is received

Reply:
sure, I can give it a try.

1. It looks like I can open a PR against that repo directly.
4. I think that we should stop the ticker after receiving the bookmark which is what we do.

Reply:
2. where do we do that?

Reply:
In line `737` (defer initialEventsEndBookmarkWarningTicker.stop()), because the reflector `exits` on `exitOnInitialEventsEndBookmark`

Reply:
I have adapted the code according to your suggestion, PTAL.